pydatastructs/linear_data_structures/arrays.py

```diff
@@ -133,13 +136,96 @@ def __len__(self):
     def __str__(self):
         return str(self._data)
 
+class MultiDimensionalArray(Array):
+    '''
+        Represents a multi-dimensional array.
+
+        Parameters
+        ==========
+
+        dtype: type
+            A valid object type.
+        *args: int
+            The dimensions of the array.
+
+        Raises
+        ======
+        IndexError
+            Index goes out of boundaries, or the number of index given is not the same as the number of dimensions
+        ValueError
+            When there's no dimensions or the dimension size is 0
+
+        Examples
+        ========
+        >>> from pydatastructs import MultiDimensionalArray as MDA
+        >>> arr = MDA(int, 5, 6, 9)
+        >>> arr.fill(32)
+        >>> arr[3, 0, 0]
+        32
+        >>> arr[3, 0, 0] = 7
+        >>> arr[3, 0, 0]
+        7
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Array_data_structure#Multidimensional_arrays
+        '''
+    __slots__ = ['_sizes', '_data', '_dtype']
+
+    def __new__(cls, dtype=NoneType, *args, **kwargs):
+        if dtype is NoneType or len(args) == (0):
+            raise ValueError("array cannot be created due to incorrect"
+                             " information.")
+        dimensions = list(args)
+        if dimensions[0] <= 1:
```

Comment:
```suggestion
        if dimensions[0] < 1:
```
`1` should be fine. Why only checked for `dimensions[0]`? Is, `dimensions[2] = 0` allowed?